### PR TITLE
[docs] Fix Pickers customization playground overflow

### LINKED ISSUE #13736: [docs] Date Range Customization Playground does not fit on smaller screens
### Related page

https://mui.com/x/react-date-pickers/playground/

### Kind of issue

Broken demo

### Issue description

image
As you can see, the date range component from the playground demo does not fit on smaller screens.
It is not possible to go to the next months on the picker, as only the Left Arrow Icon Button is visible.

### Context

Information for my screen is the following:
image

**Search keywords**: picker customization

### PR DESCRIPTION
**Pull request by LukasTy**
Fixes #13736.

Also add `DateTimeRangePicker` into this playground.

### CODE REVIEW

File: docs/src/modules/components/PickersPlayground.tsx [hunk 1]
```diff
@@ -392,6 +396,7 @@ export default function PickersPlayground() {
       ampm: ampm !== undefined ? ampm : undefined,
       ampmInClock: ampmInClock !== undefined ? ampmInClock : undefined,
       displayStaticWrapperAs: isStaticDesktopMode ? 'desktop' : 'mobile',
+      sx: { [`&.${pickersLayoutClasses.root}`]: { overflowX: 'auto' } },
```
Comment (LukasTy) [inline docs/src/modules/components/PickersPlayground.tsx:399]: This is the only relevant bugfix.

**Comment by TheOneTheOnlyJJ** [comment]
For some reason the new `DateTimeRangePicker` does not show at all when selected:
image

**Comment by LukasTy** [comment]
> For some reason the new `DateTimeRangePicker` does not show at all when selected

It does. The Desktop and Mobile variants are rendered.
It's only the Static that is not present, because we did not create this version. at least initially. 🤔

**Review by arthurbalduini — APPROVED** [review]
Nice improvement ! 🙂

**Pull request merged**